src/libraries/System.Text.Json/tests/JsonElementParseTests.cs

```diff
@@ -0,0 +1,142 @@
+                yield return new object[] { "[]", JsonValueKind.Array };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParseCases))]
+        public static void ParseValue(string json, JsonValueKind kind)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json));
+
+            JsonElement? element = JsonElement.ParseValue(ref reader);
+            Assert.Equal(kind, element!.Value.ValueKind);
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParseCases))]
+        public static void TryParseValue(string json, JsonValueKind kind)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json));
+
+            bool success = JsonElement.TryParseValue(ref reader, out JsonElement? element);
+            Assert.True(success);
+            Assert.Equal(kind, element!.Value.ValueKind);
+        }
+
+        public static IEnumerable<object[]> ElementParsePartialDataCases
+        {
+            get
+            {
+                yield return new object[] { "\"MyString"};
+                yield return new object[] { "{" };
+                yield return new object[] { "[" };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParsePartialDataCases))]
+        public static void ParseValuePartialDataFail(string json)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json));
+
+            try
+            {
+                JsonElement.ParseValue(ref reader);
+                Assert.True(false, "Expected exception.");
+            }
+            catch (JsonException) { }
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParsePartialDataCases))]
+        public static void TryParseValuePartialDataFail(string json)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json));
+
+            try
+            {
+                JsonElement.TryParseValue(ref reader, out JsonElement? element);
+                Assert.True(false, "Expected exception.");
+            }
+            catch (JsonException) { }
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParsePartialDataCases))]
+        public static void ParseValueOutOfData(string json)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json), isFinalBlock: false, new JsonReaderState());
+
+            try
+            {
+                JsonElement.ParseValue(ref reader);
+                Assert.True(false, "Expected exception.");
+            }
+            catch (JsonException) { }
+        }
+
+        [Theory]
+        [MemberData(nameof(ElementParsePartialDataCases))]
+        public static void TryParseValueOutOfData(string json)
+        {
+            var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(json), isFinalBlock: false, new JsonReaderState());
+            Assert.False(JsonElement.TryParseValue(ref reader, out JsonElement? element));
```

Comment:
Verify that `JsonElement` is still default/it's value kind is `Undefined`.

Additionally, maybe pass in an already initialized `JsonElement` and validate it doesn't change when `TryParseValue` returns false.

Reply:
Since a `Nullable` is returned, it will be null. I'll add a check for that.